src/core/blueprint/into_schema.rs

```diff
@@ -100,13 +102,55 @@ fn to_type(def: &Definition) -> dynamic::Type {
                         //                HOT CODE STARTS HERE
                         // --------------------------------------------------
 
+                        fn perform_rename(
+                            path: Vec<String>,
+                            data: &ConstValue,
+                            renames: &HashMap<Vec<String>, String>,
+                        ) -> ConstValue {
+                            match data {
+                                ConstValue::Object(obj) => {
+                                    let mut new_object = IndexMap::new();
+                                    for (name, item) in obj.into_iter() {
+                                        let mut key = path.clone();
+                                        key.push(name.to_string());
+                                        if let Some(rename) = renames.get(&key) {
+                                            new_object.insert(
+                                                Name::new(rename),
+                                                perform_rename(key, item, renames),
+                                            );
+                                        } else {
+                                            new_object.insert(
+                                                name.clone(),
+                                                perform_rename(key, item, renames),
+                                            );
+                                        }
+                                    }
+                                    ConstValue::Object(new_object)
+                                }
+                                _ => data.clone(),
+                            }
+                        }
+
+                        let args: ConstValue = {
+                            let mut new_map = IndexMap::new();
+                            let index_map = ctx.args.as_index_map();
+                            for arg in &field_copy.args {
+                                let renames = &arg.renames;
+                                let arg_key = Name::new(&arg.name);
+                                if let Some(data) = index_map.get(&arg_key) {
+                                    let data = perform_rename(vec![], data, renames);
+                                    new_map.insert(arg_key, data);
+                                }
+                            }
+                            ConstValue::Object(new_map)
+                        };
```

Comment:
This logic needs to move into `IR` under a special variant perhaps - `IR::ModifyArgs`